ReactNativeClient/lib/models/Setting.js

```diff
@@ -652,17 +656,32 @@ class Setting extends BaseModel {
 		return this.modelSelectAll('SELECT * FROM settings').then(rows => {
 			this.cache_ = [];
 
-			for (let i = 0; i < rows.length; i++) {
-				const c = rows[i];
+			const pushItemsToCache = (items) => {
+				for (let i = 0; i < items.length; i++) {
+					const c = items[i];
 
-				if (!this.keyExists(c.key)) continue;
-				c.value = this.formatValue(c.key, c.value);
-				c.value = this.filterValue(c.key, c.value);
+					if (!this.keyExists(c.key)) continue;
 
-				this.cache_.push(c);
-			}
+					c.value = this.formatValue(c.key, c.value);
+					c.value = this.filterValue(c.key, c.value);
+
+					this.cache_.push(c);
+				}
+			};
 
-			this.dispatchUpdateAll();
+			pushItemsToCache(rows);
+
+			shim.loadSecureItems(this.constants_.appId).then(secureItems => {
+				// secureItems is an array of { account: 'foo', password: 'bar' }
+				// but each item should be in the form of { key: 'foo', value: 'bar' }
+				// https://stackoverflow.com/a/50101979
+				for (const item of secureItems) {
+					delete Object.assign(item, { ['key']: item['account'] })['account'];
+					delete Object.assign(item, { ['value']: item['password'] })['password'];
+				}
+				pushItemsToCache(secureItems);
+				this.dispatchUpdateAll();
+			});
```

Comment:
The logic is not right here because the `load()` function will exit before your code here has run, which could cause race conditions. I think you should change this to `return shim.loadSecureItems(...` although ideally all this method should be refactored with async/await.

Reply:
I hadn't considered that. I will refactor them with async/await.